### PR TITLE
Use disallowed list instead of blacklist and allow list instead of whitelist

### LINKED ISSUE #3159: Change `blocklist` instead `blacklist`, `allowlist` instead `whitelist`
For us, changing this can be the correct direction.

See https://gitlab.com/gitlab-org/gitlab/-/issues/7554
See https://github.com/mochajs/mocha/pull/4355

### PR DESCRIPTION
**Pull request by heowc**
Motivation:

Related #3159

Modifications:

- Use `disallowedList` instead of `blacklist`
- Use `allowedList` instead of `whitelist`

Result:

Closed #3159

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java [hunk 1]
```diff
@@ -139,82 +139,82 @@ public boolean equals(AsciiString a, AsciiString b) {
...
     static {
-        HTTP_TO_HTTP2_HEADER_BLACKLIST.add(HttpHeaderNames.CONNECTION, EMPTY_STRING);
-        HTTP_TO_HTTP2_HEADER_BLACKLIST.add(HEADER_NAME_KEEP_ALIVE, EMPTY_STRING);
-        HTTP_TO_HTTP2_HEADER_BLACKLIST.add(HEADER_NAME_PROXY_CONNECTION, EMPTY_STRING);
-        HTTP_TO_HTTP2_HEADER_BLACKLIST.add(HttpHeaderNames.TRANSFER_ENCODING, EMPTY_STRING);
-        HTTP_TO_HTTP2_HEADER_BLACKLIST.add(HttpHeaderNames.UPGRADE, EMPTY_STRING);
-        HTTP_TO_HTTP2_HEADER_BLACKLIST.add(ExtensionHeaderNames.STREAM_ID.text(), EMPTY_STRING);
-        HTTP_TO_HTTP2_HEADER_BLACKLIST.add(ExtensionHeaderNames.SCHEME.text(), EMPTY_STRING);
-        HTTP_TO_HTTP2_HEADER_BLACKLIST.add(ExtensionHeaderNames.PATH.text(), EMPTY_STRING);
+        HTTP_TO_HTTP2_HEADER_BLOCKLIST.add(HttpHeaderNames.CONNECTION, EMPTY_STRING);
```
Comment (anuraaga) [inline core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java:155]: Somehow block list seems a bit different - blocking requests with these headers is what came to mind first. Perhaps `FILTERED_*` is a good pattern?

Comment (heowc) [reply core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java:155]: It seems like a better idea. 😄

File: core/src/main/java/com/linecorp/armeria/client/IgnoreHostsTrustManager.java [hunk 2]
```diff
@@ -49,7 +49,7 @@
 import com.google.common.collect.ImmutableSet;
 
 /**
- * An implementation of {@link X509ExtendedTrustManager} that skips verification on whitelisted hosts.
+ * An implementation of {@link X509ExtendedTrustManager} that skips verification on allowlisted hosts.
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/client/IgnoreHostsTrustManager.java:52]: There is no verb 'allowlist'. How about: the list of allowed hosts?

File: core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java (hunk 1 above)
Comment (trustin) [reply core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java:155]: `DISALLOWED_` also sounds good to me.

File: core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java [hunk 3]
```diff
@@ -881,11 +881,11 @@ private static void toNettyHttp2Client(HttpHeaders inputHeaders, Http2Headers ou
         for (Entry<AsciiString, String> entry : inputHeaders) {
             final AsciiString name = entry.getKey();
             final String value = entry.getValue();
-            if (HTTP_TO_HTTP2_HEADER_BLACKLIST.contains(name)) {
+            if (HTTP_TO_HTTP2_HEADER_BLOCKLIST.contains(name)) {
                 continue;
             }
 
-            if (isTrailer && isTrailerBlacklisted(name)) {
+            if (isTrailer && isTrailerBlocklisted(name)) {
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java:888]: Blocklisted -> Disallowed?

File: core/src/test/java/com/linecorp/armeria/client/ClientOptionsTest.java [hunk 4]
```diff
@@ -66,7 +66,7 @@ void testSetHeader() {
     }
 
     @Test
-    void testSetBlackListHeader() {
+    void testSetBlockListHeader() {
```
Comment (trustin) [inline core/src/test/java/com/linecorp/armeria/client/ClientOptionsTest.java:69]: testSetDisallowedHeader ?

**Review by trustin — COMMENTED** [review]
It'd be better using allowed/disallowed instead of allowlisted/blocklisted.

File: core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java (hunk 3 above)
Comment (KarboniteKream) [reply core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java:888]: Forbidden also sounds good, but I'm not sure if it's too "strict".

Comment (trustin) [reply core/src/main/java/com/linecorp/armeria/internal/common/ArmeriaHttpUtil.java:888]: That sounds good to me as well.

File: core/src/main/java/com/linecorp/armeria/client/IgnoreHostsTrustManager.java (hunk 2 above)
Comment (heowc) [reply core/src/main/java/com/linecorp/armeria/client/IgnoreHostsTrustManager.java]: Fixed

**Comment by heowc** [comment]
I changed most of the `blocklist` to the `disallowed list`. 😄

**Comment by minwoox** [comment]
> I changed most of the blocklist to the disallowed list. 

How about changing all of them to disallowed list? 😄

**Comment by heowc** [comment]
> > I changed most of the blocklist to the disallowed list.
> 
> How about changing all of them to disallowed list? 😄

Yes! I Fixed. 😉

**Review by minwoox — APPROVED** [review]
Thanks, @heowc!

**Comment by minwoox** [comment]
Please fix the check style. 😄 
https://ci.appveyor.com/project/line/armeria/builds/36206039/job/kb29ur4vb0xbx8rx#L536

**Comment by heowc** [comment]
> Please fix the check style. 😄
> https://ci.appveyor.com/project/line/armeria/builds/36206039/job/kb29ur4vb0xbx8rx#L536

Oops... I fixed 😆

**Review by ikhoon — APPROVED** [review]
Thanks! @heowc

**Review by trustin — APPROVED** [review]

**Pull request merged**